src/crypto/kms.rs

```diff
@@ -0,0 +1,237 @@
+//! KMS implementation for cryptography
+
+use openssl::{
+    bn::BigNum,
+    ecdsa::EcdsaSig,
+    hash::MessageDigest,
+    pkey::{PKey, Public},
+};
+use tokio::runtime::Runtime;
+
+use aws_sdk_kms::{
+    error::{VerifyError, VerifyErrorKind},
+    model::{MessageType, SigningAlgorithmSpec},
+    Blob, Client, SdkError,
+};
+
+use crate::{
+    crypto::{ec_curve_to_parameters, SigningPrivateKey, SigningPublicKey},
+    error::CoseError,
+    sign::SignatureAlgorithm,
+};
+
+/// A reference to an AWS KMS key and client
+pub struct KmsKey {
+    client: Client,
+    key_id: String,
+
+    sig_alg: SignatureAlgorithm,
+
+    public_key: Option<PKey<Public>>,
+
+    runtime: Runtime,
+}
+
+impl KmsKey {
+    fn new_runtime() -> Runtime {
```

Comment:
Creating a runtime in a library is super weird/anti-pattern.
Why not make your functions async and leave the runtime stuff to the application calling this library?

Reply:
I need to take another look, but it required making the methods async, which meant it didn't fit with any of the other interfaces. Maybe there's a way around this, but I'm not sure what that is.

Reply:
It's solved in the latest commit.